### DISCUSSION TITLE
Lambda is going into retry loop in same invocation.

### DISCUSSION THREAD

**Discussion by TOPSinfo** [brefphp/bref#1045] [Support]
**Description:**

(Within the Same invocation ) You will find “Trigger Started” printed more than 1000 times. It looks like lambda is retrying within the same invocation which we are trying to stop. 

Can anyone please help here?

![image](https://user-images.githubusercontent.com/30435005/135438603-bd0420d8-6ee3-40bb-ae26-959fe29e18f1.png)

**Comment by mnapoli** [comment]
Moving to a discussion as this is support.

For the record it's almost impossible to help you without more details. My guess is that you use the function runtime and your code `exit()` somewhere, but that's just a wild guess.

**Reply by TOPSinfo** [reply]
Sorry for not being able to give enough details here as my problem is pretty weird.

I didn't get what you are guessing here, are you saying is that my code is breaking in between because of a function error which is resulting in multiple retries within the same invocation?

**Reply by mnapoli** [reply]
Share serverless.yml, that's the only way we can help.

**Comment by deleugpn** [comment]
I'm guessing you have a loop inside your own code. As you can see there's no START REQUESTID and END REQUESTID between logs which seems like you're inside just a single invocation.

**Reply by TOPSinfo** [reply]
One thing is certain that it's happening within a single invocation. But the loop is not inside the code, It seems like some kind of retry at lambda level when there is any error from runtime.  Wondering if this might be possible in the case of PHP runtime, coz this is random and not happening on every invocation.

**Reply by deleugpn** [reply]
There is no lambda retry. If there was, you'd see `START REQUESTID` and `END REQUESTID` for each retry.

**Reply by TOPSinfo** [reply]
Is it possible that something in bref is being called recursively ?

It's my serverless laravel application and that print statement on the screenshot above is from worker.php which is a starting point of execution. 
<img width="1205" alt="Screenshot 2021-10-01 at 11 56 46 AM" src="https://user-images.githubusercontent.com/30435005/135575586-0a94900d-5dde-4a1b-a00b-3634b15575d0.png">

**Reply by mnapoli** [reply]
@deleugpn I'm guessing they use the function runtime with an exit -> 1 invocation, but Lambda keeps trying to restart the PHP process because of `exit` (it thinks the process crashed).

But that's just a guess.

**Reply by TOPSinfo** [reply]
I also think the same. That lambda keeps trying to restart PHP until it reaches the timeout limit of 15 minutes. Is there any possible configuration where we can stop this from happening?

**Reply by mnapoli** [reply]
Don't use `exit()`.

**Reply by akshit-arora** [reply]
> 
> 
> Don't use `exit()`.

Thank you so much. This helped in fixing out that recursive call.
I had applied the `exit()` whenever I was checking if the process is already running to make sure it runs once at a time. Now i have applied `dd("Process already running")` but my cloudwatch logs show it at runtime exit with error.

> Error: Runtime exited with error: exit status 1 Runtime.ExitError

Can you please suggest a command by which I can gracefully stop the execution of php script for this case?

**Reply by TOPSinfo** [reply]
Not using exit helped. Thank you so much for the help. It was much needed.

**Reply by mnapoli** [reply]
> Can you please suggest a command by which I can gracefully stop the execution of php script for this case?

The handler should use `return` (to stop the execution of a Lambda invocation).
